src/pages/dashboard/analysis/components/Charts/Bar/index.tsx

```diff
@@ -0,0 +1,133 @@
+import { Axis, Chart, Geom, Tooltip } from 'bizcharts';
+import React, { Component } from 'react';
+
+import Debounce from 'lodash.debounce';
+import autoHeight from '../autoHeight';
+import styles from '../index.less';
+
+export interface BarProps {
+  title: React.ReactNode;
+  color?: string;
+  padding?: [number, number, number, number];
+  height?: number;
+  data: {
+    x: string;
+    y: number;
+  }[];
+  forceFit?: boolean;
+  autoLabel?: boolean;
+  style?: React.CSSProperties;
+}
+
+class Bar extends Component<
+  BarProps,
+  {
+    autoHideXLabels: boolean;
+  }
+> {
+  state = {
+    autoHideXLabels: false,
+  };
+
+  root: HTMLDivElement | undefined = undefined;
+
+  node: HTMLDivElement | undefined = undefined;
+
+  resize = Debounce(() => {
+    if (!this.node || !this.node.parentNode) {
+      return;
+    }
+    const canvasWidth = (this.node.parentNode as HTMLDivElement).clientWidth;
+    const { data = [], autoLabel = true } = this.props;
+    if (!autoLabel) {
+      return;
+    }
+    const minWidth = data.length * 30;
+    const { autoHideXLabels } = this.state;
+
+    if (canvasWidth <= minWidth) {
+      if (!autoHideXLabels) {
+        this.setState({
+          autoHideXLabels: true,
+        });
+      }
+    } else if (autoHideXLabels) {
+      this.setState({
+        autoHideXLabels: false,
+      });
+    }
+  }, 500);
+
+  componentDidMount() {
+    window.addEventListener('resize', this.resize, { passive: true });
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('resize', this.resize);
+  }
+
+  handleRoot = (n: HTMLDivElement) => {
+    this.root = n;
+  };
+
+  handleRef = (n: HTMLDivElement) => {
+    this.node = n;
+  };
+
+  render() {
+    const {
+      height = 1,
+      title,
+      forceFit = true,
+      data,
+      color = 'rgba(24, 144, 255, 0.85)',
+      padding,
+    } = this.props;
+
+    const { autoHideXLabels } = this.state;
+
+    const scale = {
+      x: {
+        type: 'cat',
+      },
+      y: {
+        min: 0,
+      },
+    };
+
+    const tooltip: [string, (...args: any[]) => { name?: string; value: string }] = [
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Type declaration of 'any' loses type-safety. Consider replacing it with a more precise type.](https://app.codacy.com/app/afc163/ant-design-pro/pullRequest?prid=4583897)